app/renderer/actions/Session.js

```diff
@@ -724,6 +727,54 @@ export function setSavedServerParams () {
   };
 }
 
+export function setStateFromAppiumFile (newFilepath = null) {
+  return (dispatch) => {
+    // no "fs" means we're not in an Electron renderer so do nothing
+    if (!fs) {
+      return;
+    }
+    try {
+      let filePath = newFilepath;
+      if (!newFilepath) {
+        const lastArg = process.argv[process.argv.length - 1];
+        if (!lastArg.startsWith('filename=')) {
+          return;
+        }
+        filePath = lastArg.split('=')[1];
+      }
+      const filePathStorageKey = 'last_opened_file';
+      if (sessionStorage.getItem(filePathStorageKey) === filePath) {
+        // file was opened already, do nothing
+        return;
+      }
+      const appiumJson = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+      sessionStorage.setItem(filePathStorageKey, filePath);
+      dispatch({type: SET_STATE_FROM_SAVED, state: appiumJson, filePath});
+    } catch (e) {
+      notification.error({
```

Comment:
shall we write the actual stack trace somewhere to the log, so it could be debugged?

Reply:
![Screen Shot 2022-04-28 at 1 01 42 PM](https://user-images.githubusercontent.com/852574/165836012-12e1938d-89f7-412b-b762-8d0f2d483679.png)

^ I'll just add it directly to the notification. It's a bit messy, but this is likely a rare case anyways because it's not going to be common for users to meddle with the file contents.

Reply:
And showing it in the notification would be helpful for users to submit bugs.